### PR TITLE
scripts: add command line args

### PR DESCRIPTION
**Pull request by fscherf**
With these patches command line arguments can be used in Lona scripts.

``python lona_script.py --port=8081 --shell -O Foo=True``

### CODE REVIEW

File: lona/app.py [hunk 1]
```diff
@@ -240,13 +333,12 @@ def setup_server(
     def run(
             self,
             loop: Union[AbstractEventLoop, None] = None,
+            parse_command_line: int = True,
```
Comment (maratori) [inline lona/app.py:336]:
```suggestion
            parse_command_line: bool = True,
```

File: lona/app.py [hunk 2]
```diff
@@ -202,17 +201,105 @@ def add_static_file(
...
+                args['settings_post_overrides'],
+            )
+
+        return args
+
     # server ##################################################################
     def setup_server(
             self,
             loop: Union[AbstractEventLoop, None] = None,
+            settings_pre_overrides: Union[Dict, None] = None,
```
Comment (maratori) [inline lona/app.py:301]: nit:
```suggestion
            settings_pre_overrides: Optional[Dict[str, Any]] = None,
```

**Review by maratori — CHANGES_REQUESTED** [review]
I believe parsing command line arguments is not a responsibility of `LonaApp`.
It already can receive arbitrary arguments in `app.run()`.

If I want to parse arguments I should do it outside of `LonaApp`.
We can add some helper function. For example
```py
app = LonaApp(__file__)
# setup views
app.run(**parse_command_line(port=8080))
```

File: lona/app.py (hunk 2 above)
Comment (fscherf) [reply lona/app.py]: Fixed

File: lona/app.py (hunk 1 above)
Comment (fscherf) [reply lona/app.py]: Fixed

File: lona/app.py [hunk 3]
```diff
@@ -257,10 +349,32 @@ def run(
...
         setup_logging(server_args)
 
+        # setup server
+        self.setup_server(
+            loop=loop,
+            settings_pre_overrides=getattr(
+                server_args,
+                'settings_pre_overrides',
+                None,
+            ),
```
[the remark this answers is not in the document]
Comment (fscherf) [reply lona/app.py]: fixed

**Comment by fscherf** [comment]
@maratori I think it's a very useful feature that you just call ``app.run()`` and can override the port for example. If we get a bug report with a reproducing script, the first thing i do is patching ``app.run()`` to ``app.run(log_level='debug', shell=True)``. It would be nice to just run it with ``-l debug --shell`` without changing something.

If we put this functionality in it's own place, i would recommend everyone to use it, because it adds the same interface to scripts like in projects, which is especially useful if you read https://lona-web.org/end-user-documentation/debugging.html while developing. I think it makes it consistent.

Also: I like debugging infrastructure that's always in place. Not only if you set it up. I can tell every user to add ``-l debug``, but not every user to patch a line of code.

Is there a reason to split that, that i miss?

**Review by maratori — APPROVED** [review]
@fscherf Ok

**Comment by fscherf** [comment]
@maratori Thanks for reviewing!

**Review by themutt — APPROVED** [review]

**Pull request merged**